libs/zzz/db/src/Database/DataManagers/CharacterOptManager.ts

```diff
@@ -140,11 +243,39 @@ export class CharacterOptManager extends DataManager<
       })
       .filter(notEmpty)
     if (!Array.isArray(bonusStats)) bonusStats = []
-    bonusStats = bonusStats.filter(({ tag, value }) => {
-      if (!validateTag(tag)) return false
-      if (typeof value !== 'number') return false
-      return true
-    })
+    bonusStats = bonusStats.map(
+      ({ tag: { q, qt, attribute, damageType1, damageType2 }, value }) => {
+        if (typeof value !== 'number') value = 0
+        q = validateValue(q, bonusStatKeys) ?? 'atk'
+        qt = validateValue(qt, bonusStatQtKeys) ?? 'combat'
+
+        if (q !== 'dmg_') attribute = undefined
+        if (attribute) attribute = validateValue(attribute, allAttributeKeys)
+
+        if (
+          !bonusStatDmgTypeIncStats.includes(
```

Comment:
fyi/nit: we also have isIn if you want. not needed, just letting you know for future